tinygrad/tensor.py

```diff
@@ -942,7 +942,7 @@ def pad(self, arg:Tuple[Optional[Tuple[sint, sint]], ...], value:float=0.0) -> T
   #   2. Bool indexing is not supported
   #   3. Out of bounds Tensor indexing results in 0
   #     - e.g: Tensor([1, 2, 3])[Tensor([4, 3, 2])] -> [0, 0, 3] index 4 and 3 are out of bounds
-  def __getitem__(self, indices) -> Tensor:
+  def __getitem__(self, indices, v: Union[Tensor, None] = None) -> Tensor:
```

Comment:
don't overload the `__getitem__` API, if you need a different function with additional Tensor as input, create a function for it.

Reply:
Will benchmark tomorrow. I can't think of a cleaner way than passing v as an argument as advanced setitem needs access to getitem's internals and v. I tried breaking down getitem but found it looked worse.

Reply:
Can you write a function they both depend on?